src/OceanSimulations/OceanSimulations.jl

```diff
@@ -9,60 +17,95 @@ using SeawaterPolynomials.TEOS10: TEOS10EquationOfState
+default_free_surface(grid) = SplitExplicitFreeSurface(grid; cfl=0.7)
 
 function default_ocean_closure()
     mixing_length = MixingLength(Cᵇ=0.01)
     turbulent_kinetic_energy_equation = TurbulentKineticEnergyEquation(Cᵂϵ=1.0)
     return CATKEVerticalDiffusivity(; mixing_length, turbulent_kinetic_energy_equation)
 end
 
+default_momentum_advection() = VectorInvariant(; vorticity_scheme = WENO(; order = 9),
+                                                  vertical_scheme = Centered(),
+                                                divergence_scheme = WENO())
+
+default_tracer_advection() = TracerAdvection(WENO(; order = 7),
+                                             WENO(; order = 7),
+                                             Centered())
+
+@inline ϕ²(i, j, k, grid, ϕ)    = @inbounds ϕ[i, j, k]^2
+@inline spᶠᶜᶜ(i, j, k, grid, Φ) = @inbounds sqrt(Φ.u[i, j, k]^2 + ℑxyᶠᶜᵃ(i, j, k, grid, ϕ², Φ.v))
+@inline spᶜᶠᶜ(i, j, k, grid, Φ) = @inbounds sqrt(Φ.v[i, j, k]^2 + ℑxyᶜᶠᵃ(i, j, k, grid, ϕ², Φ.u))
+
+@inline u_quadratic_bottom_drag(i, j, grid, c, Φ, μ) = @inbounds - μ * Φ.u[i, j, 1] * spᶠᶜᶜ(i, j, 1, grid, Φ)
+@inline v_quadratic_bottom_drag(i, j, grid, c, Φ, μ) = @inbounds - μ * Φ.v[i, j, 1] * spᶜᶠᶜ(i, j, 1, grid, Φ)
+
+@inline u_immersed_quadratic_bottom_drag(i, j, k, grid, c, Φ, μ) = @inbounds - μ * Φ.u[i, j, k] * spᶠᶜᶜ(i, j, k, grid, Φ)
+@inline v_immersed_quadratic_bottom_drag(i, j, k, grid, c, Φ, μ) = @inbounds - μ * Φ.v[i, j, k] * spᶜᶠᶜ(i, j, k, grid, Φ)
+
+@inline is_immersed_drag_u(i, j, k, grid) = Int(immersed_peripheral_node(i, j, k-1, grid, Face(), Center(), Center()) & !inactive_node(i, j, k, grid, Face(), Center(), Center()))
+@inline is_immersed_drag_v(i, j, k, grid) = Int(immersed_peripheral_node(i, j, k-1, grid, Center(), Face(), Center()) & !inactive_node(i, j, k, grid, Center(), Face(), Center()))
+
+# Keep a constant linear drag parameter independent on vertical level
+@inline u_immersed_bottom_drag(i, j, k, grid, clock, fields, μ) = @inbounds - μ * fields.u[i, j, k] * is_immersed_drag_u(i, j, k, grid) * spᶠᶜᶜ(i, j, k, grid, fields) / Δzᶠᶜᶜ(i, j, k, grid)
+@inline v_immersed_bottom_drag(i, j, k, grid, clock, fields, μ) = @inbounds - μ * fields.v[i, j, k] * is_immersed_drag_v(i, j, k, grid) * spᶜᶠᶜ(i, j, k, grid, fields) / Δzᶜᶠᶜ(i, j, k, grid)
+
 # TODO: Specify the grid to a grid on the sphere; otherwise we can provide a different
 # function that requires latitude and longitude etc for computing coriolis=FPlane...
-function ocean_simulation(grid;
+function ocean_simulation(grid; Δt = 5minutes,
                           closure = default_ocean_closure(),
                           free_surface = default_free_surface(grid),
                           reference_density = 1020,
                           rotation_rate = Ω_Earth,
-                          gravitational_acceleration = g_Earth)
+                          gravitational_acceleration = g_Earth,
+                          drag_coefficient = 0.003,
```

Comment:
```suggestion
                          bottom_drag_coefficient = 0.003,
```

?

We may want to implement something neater eventually